### PR TITLE
feat: auto WebGL terminal renderer with gpuAcceleration setting

### PR DESCRIPTION
**Pull request by sam-phinizy**
## Summary

- Terminal panes now recover from WebGL context loss (GPU crash, suspended tab, too many WebGL contexts) by disposing the WebGL addon and reverting to xterm's built-in DOM renderer — no pane drop, no scrollback loss.
- New **GPU acceleration** dropdown on the Terminal settings tab (`Auto` / `On (WebGL)` / `Off (DOM)`), modeled on VSCode's `terminal.integrated.gpuAcceleration`. Setting changes apply to terminals opened after the change; existing panes keep their renderer until reopened.
- `Auto` is the default and is identical to today's behavior on the happy path; the new value of this PR is graceful recovery + an opt-out for users on systems where WebGL is flaky.

## Implementation notes

- `crates/roux-core/src/models/settings.rs` — new `GpuAcceleration` enum + `gpu_acceleration` field on `RouxSettings`. `#[serde(default)]` means existing settings files load unchanged.
- `src/lib/panes/xtermController.ts` — replaced the unconditional `try { loadAddon(new WebglAddon()) }` with a `setupRenderer(mode)` helper that wires `WebglAddon.onContextLoss` for runtime recovery and tracks the addon for explicit disposal.
- `src/lib/components/SettingsPanel.svelte` — added the dropdown to the Terminal tab.
- `src/lib/bindings.ts` / `src/lib/types.ts` — generated type added (will be regenerated identically on the next debug build) plus new field in the hand-maintained `DEFAULT_SETTINGS`.
- Docs updated: `docs/settings.md` (new GPU acceleration subsection) and `docs/whats-new.md` (April 28 entry).

### About the fallback chain

Initial plan was WebGL → Canvas → DOM (matching VSCode), but `@xterm/addon-canvas` has no version compatible with `@xterm/xterm@6` (peer dep `^5.0.0` on every published version, and the addon was dropped from xterm.js master). Pivoted to WebGL → DOM only, which is what xterm@6 natively supports. The user-facing setting semantics are unchanged.

## Test plan

- [x] `cargo check -p roux-core` — green
- [x] `npm run check` — 0 errors, 0 warnings, 897 files
- [x] `npm run test` — 64 files / 735 passed (2 pre-existing skips), including 8 new tests in `src/lib/panes/__tests__/xtermController.test.ts` covering `auto`/`on`/`off`, missing-field default, WebGL construction failure, context-loss disposal, and double-dispose safety
- [ ] Manual: open a Claude pane, confirm `<canvas data-type="webgl">` is present
- [ ] Manual: switch GPU acceleration to **Off**, open a new pane, confirm DOM renderer (no `data-type=webgl` canvas)
- [ ] Manual: with WebGL active, run in DevTools console: `document.querySelector('.xterm canvas[data-type=webgl]').getContext('webgl2').getExtension('WEBGL_lose_context').loseContext()` — terminal should keep working and the WebGL canvas should be gone

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a Terminal GPU acceleration setting (Auto default, On, Off) to choose WebGL or DOM rendering.
  * Terminals now gracefully fall back to DOM rendering if WebGL is unavailable or the context is lost; setting takes effect for newly opened panes.

* **Documentation**
  * Updated settings docs and release notes to describe the GPU acceleration options and behavior.

* **Tests**
  * Added tests verifying WebGL initialization, context-loss handling, disposal, and fallback behavior.

### CODE REVIEW

File: src/lib/panes/xtermController.ts [hunk 1]
```diff
@@ -54,6 +52,28 @@ class XtermTerminalController implements TerminalController {
...
+    // dispose the addon so xterm reverts to its built-in DOM renderer.
+    try {
+      const webgl = new WebglAddon();
+      this.terminal.loadAddon(webgl);
+      this.webglAddon = webgl;
+      webgl.onContextLoss(() => {
+        webgl.dispose();
+        if (this.webglAddon === webgl) {
+          this.webglAddon = null;
+        }
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply src/lib/panes/xtermController.ts]: Fixed in 5531db3. Reordered the guard so the callback checks `this.webglAddon !== webgl` first and nulls the ref before calling `webgl.dispose()`, making it safe against re-entrant or late events. Also captured the `IDisposable` returned by `onContextLoss` and dispose it in `XtermTerminalController.dispose()` to prevent the handler firing after teardown.

File: src/lib/panes/__tests__/xtermController.test.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+    const controller = createXtermTerminalController();
+    mocks.lastWebglContextLossHandler?.();
+    expect(mocks.webglDispose).toHaveBeenCalledTimes(1);
+
+    controller.dispose();
+
+    expect(mocks.webglDispose).toHaveBeenCalledTimes(1);
+    expect(mocks.terminalDispose).toHaveBeenCalledTimes(1);
+  });
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply src/lib/panes/__tests__/xtermController.test.ts:198]: Fixed in 5531db3. Added three regression tests covering the edge cases: 'disposes the onContextLoss subscription when the controller is disposed', 'ignores a context-loss event that fires after the controller is disposed', and 'only disposes the WebglAddon once if onContextLoss fires twice'.
